File: programmers/Hash/3.js
function solution(clothes) {
    var answer = 0;
    const type={};
    clothes.forEach((el)=>{
       type[el[1]]=(type[el[1]]|0)+1;
    });
    const multi=Object.values(type);
    let temp=1;
    while(multi.length){
       temp*=(multi.pop()+1) 
    }
    temp-=1;
    answer=temp
    return answer;
}

//https://programmers.co.kr/learn/courses/30/lessons/42578

File: programmers/Greedy/3.py
#https://programmers.co.kr/learn/courses/30/lessons/42883

def solution(number, k):
    answer = ''
    result=[number[0]]
    i=0
    for i in number[1:]:
        while len(result)>0and result[-1]<i  and k>0:
            result.pop()
            k-=1
        result.append(i)            
    if k!=0:
        result=result[:-k]
    answer=''.join(result)
    return answer
    
